DXMainClient/DXGUI/Generic/LoadingScreen.cs

```diff
@@ -71,6 +72,36 @@ public override void Initialize()
             }
         }
 
+        protected override void GetINIAttributes(IniFile iniFile)
+        {
+            base.GetINIAttributes(iniFile);
+
+            string value = iniFile.GetStringValue(Name, "RandomTextures", string.Empty).Trim();
+
+            randomTextures = value == string.Empty ? null : value.Split(',', StringSplitOptions.RemoveEmptyEntries).ToList();
+            
+            int len = randomTextures.Count;
+
+            for (int i = 0; i < len; i++)
+            {
+                randomTextures[i] = randomTextures[i].Trim();
+                if (randomTextures[i] == string.Empty)
+                {
+                    randomTextures.RemoveAt(i);
+                    len -= 1;
+                }
+            }
```

Comment:
This whole thing should be further refactored into something like `SplitAndTrim`. The bigger issue is that there's an error here: if you encounter an entry to be deleted and delete it -- you will also need to decrement `i` too.
```suggestion
            for (int i = 0; i < len; i++)
            {
                randomTextures[i] = randomTextures[i].Trim();
                if (randomTextures[i] == string.Empty)
                {
                    randomTextures.RemoveAt(i);
                    len--;
                    i--;
                }
            }
```

I would also suggest to not use `StringSplitOptions.RemoveEmptyEntries` because you effectively do that by yourself. `Trim()` call and your check will achieve that.

Reply:
True. Didn't think of it.
`.Select(s => s.Trim()).Where(s => !s.IsNullOrEmpty())`

Reply:
`Where(string.IsNullOrEmpty)`

There can't be a non-static method like `s.IsNull()` because there will be an null reference exception :)

Reply:
@SadPencil @Metadorius check e818ce03d1a3dd9446c375dd0a4126a8221d6a07.

Reply:
I'll modify the codes for you later.

```
string value = iniFile.GetStringValue(Name, "RandomTextures", string.Empty).Trim(); // The trim here makes no sence

randomTextures = value == string.Empty ? null; // you don't have to determining whether the string is null or not when using this linq expression. Split() works on empty string.
```